solutions/usaco-717.mdx

```diff
@@ -2,14 +2,84 @@
 id: usaco-717
 source: USACO Gold 2017 December
 title: Why Did the Cow Cross the Road
-author: Qi Wang
+author: Qi Wang, Jeffrey Zhang
 ---
 
 [Official Analysis](http://www.usaco.org/current/data/sol_visitfj_gold_feb17.html)
 
 ## Implementation
 
 <LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+typedef long long ll;
+typedef pair<int,int> pii;
+
+//Possible directions a cow can go to eat grass
+int Dx[] = {1, 0, -1, 0, 3, 0, -3, 0, 2, 2, 1, 1, -1, -1, -2, -2};
+int Dy[] = {0, 1, 0, -1, 0, 3, 0, -3, 1, -1, 2, -2, 2, -2, 1, -1};
+
+int main() {
+    ifstream fin("visitfj.in");
+    ofstream fout("visitfj.out");
+
+    int n, t;
+    fin >> n >> t;
+    vector<vector<int>> field(n, vector<int>(n));
+    for (int i = 0; i < n; i++){
+        for (int j = 0; j < n; j++){
+            fin >> field[i][j];
+        }
+    }
+
+    /* 
+     * There are 16 possible tiles a cow can reach in 3 moves
+     * Form edges between those tiles and with assigned weight:
+     * (the time it takes to eat the tile) + 3 * t
+     * If the tile is less than 3 manhattan distance to farmer John's tile,
+     * form an edge between those two tiles also.
+     * The problem is now simple dijkstra
+     */
+    vector<vector<int>> dist(n, vector<int>(n, INT_MAX));
+    vector<vector<bool>> visited(n, vector<bool>(n));
+    dist[0][0] = 0;
+    priority_queue<pair<int, pii>> dijkstra;
+    dijkstra.push(make_pair(0, make_pair(0,0)));
+    while (dijkstra.size()){
+        int x, y;
+        tie(x, y) = dijkstra.top().ss;
+        dijkstra.pop();
+        if (visited[x][y]){
+			continue;
+		}
+        visited[x][y] = true;
+        for (int i = 0; i < 16; i++){
+            int dx = x + Dx[i], dy = y + Dy[i];
```

Comment:
just something minor:
`dx` typically stands for delta x, which means change in x. However `dx` in this case is the actual new x. I would recommend using something like `newx` here and decapitalize the array variable because it can get confusing

Reply:
Ok good idea, I will address that